dev/PushNotifications/PushNotificationChannel.h

```diff
@@ -29,6 +30,17 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         // IWpnForegroundSink
         HRESULT __stdcall InvokeAll(_In_ ULONG length, _In_ byte* payload, _Out_ BOOL* foregroundHandled) noexcept;
 
+        //INotificationHandler
+        HRESULT __stdcall OnRawNotificationReceived(unsigned int payloadLength, _In_ byte* payload, _In_ HSTRING /*correlationVector */) noexcept
+        {
+            BOOL foregroundHandled = false;
+            THROW_IF_FAILED(InvokeAll(payloadLength, payload, &foregroundHandled));
+            THROW_HR_IF(E_NOT_VALID_STATE, !foregroundHandled);
```

Comment:
In this scenario, when the foreground isn't handled, we need to find a way to route the args to background activation. So that means protocol activation will have to happen here 😄 . Check the LRP to see how protocol activation is currently being done. Maybe start a commons file shared between LRP/WAS.

Reply:
Correct. If the foreground event is not handled, we should re-launch the app with the payload using Protocol. That would mean that this line of code needs to go away:
 THROW_HR_IF(E_NOT_VALID_STATE, !foregroundHandled);

Reply:
Also please test this corner case scenario : )

Reply:
I agree to the fact that we should not have the line set.  Firslty, let me reiterate we are dealing with packaged applications. Let me put the scenarios here:

1. Foreground handled - args.Cancel is set to true - this will avoid background task to be triggered
2. Foreground not handled - args.Cancel is set to false - then if the background task is also registered by the user then backgroundtask is triggered.

The above cases are handled if we remove the foregroundhandled check. Do we really need the protocol activation here.?

Reply:
I'm not sure I understand the question. The contract is straightforward here: If the developer chooses not to intercept the foreground event, then the expectation is that the background trigger is executed. In this case, there is a noop. So it is a bug : )

Reply:
I also think that packaged apps using BI suffer the same limitation here. We should file a seperate bug for that and fix it as well.

Reply:
https://github.com/microsoft/WindowsAppSDK/issues/1490